lib/Dropdown.js

```diff
@@ -0,0 +1,775 @@
+			if (this.#pen.frameCount % inverseScrollSpeed === 0){
+				this.#moveIndexUpWhen("down")
+			}
+			return;
+		} else if (this.#pen.keys.durationDown("uparrow") > this.#arrowHoldDelay) {
+			if (this.#pen.frameCount % inverseScrollSpeed === 0){
+				this.#moveIndexUpWhen("up")
+			}
+			return;
+		}
+		// Pressing arrow keys
+		if (this.#pen.keys.released("downarrow")){
+			this.#moveIndexUpWhen("down");
+		} else if (this.#pen.keys.released("uparrow")){
+			this.#moveIndexUpWhen("up")
+		} 
+	}
+	/**
+	 * Moves index up or down, depending on the direction given
+	 * @param {string} direction "up" or "down"
+	 */
+	#moveIndexUpWhen(direction) {
+		if(this.#openDirection === direction) {
+			if (this.#focussedIndex + 1 <= this.#options.length - 1) {
+				this.#focussedIndex++;
+			}
+		} else {
+			if (this.#focussedIndex - 1 >= 0) {
+				this.#focussedIndex--;
+			}
+		}
+	}
+	/**
+	 * Selects the focussed option and closes the dropdown when enter is pressed
+	 */
+	#handleEnterInput(){
+		// Check if the draw command is issued after logic
+		// Prevents issues where dropdown is non-responsive
+		if (
+			this.#lastFrameDrawn !== this.#pen.frameCount - 1 &&
+			this.#lastFrameDrawn !== this.#pen.frameCount
+		) {
+			return;
+		}
+
+		if (this.#pen.keys.released("enter")) {
+			this.#index = this.#focussedIndex;
+			this.#open = false;
+		}
+	}
+    /**
+     * Returns true if the given x and y coordinates are within the bounds of the given rectangle.
+     * @param {number} x
+     * @param {number} y
+     * @param {number} centerX
+     * @param {number} centerY
+     * @param {number} width
+     * @param {number} height
+     * @returns {boolean}
+     * @static
+     */
+    static #isInRect(x, y, centerX, centerY, width, height) {
+        const leftX = centerX - width / 2;
+        const topY = centerY - height / 2;
+        return (
+            x >= leftX && x <= leftX + width && y >= topY && y <= topY + height
+        );
+    }
+	/**
+	 * Handles type and condition checking for the options parameter. Must be an array of strings.
+	 * @param {string[]} options 
+	 * @throws {Error} Throws error if what is given is not an array, or if any items inside the array are not strings.
+	 */
+    static checkOptionsConditions(options) {
+        if (Array.isArray(options) === false) {
+            throw Error(
+                `You need to give an array for the options.\n` + 
+                    `You gave: ${options}: ${typeof options} \n\n` +
+                    `For example, it is acceptable to give:\n` + 
+                    `   ["Option 1", "Option 2", "Option 3"]`
+            )
+        }
+        let arrayTypeIssues = ``;
+        let errorFound = false;
+        for (let i = 0; i < options.length; i++) {
+            if (typeof options[i] !== "string") {
+                arrayTypeIssues += `  ❌ At index ${i}, you gave: ${options[i]} which is a ${typeof options[i]}.\n`
```

Comment:
There is a method printing out a thin preview of the contents of an array in the errorMsgManager that could be useful here potentially?  Up to you though if you think it'd be helpful not too worried either way.

Reply:
From the beginning I was told to ignore the error message class (assuming it would be implemented in the GUI later?) so to keep it consistent I might just leave it for now